pkg/simple/client/devops/jenkins/internal/gitlab.go

```diff
@@ -0,0 +1,154 @@
+package internal
+
+import (
+	"github.com/beevik/etree"
+	devopsv1alpha3 "kubesphere.io/kubesphere/pkg/apis/devops/v1alpha3"
+	"strconv"
+	"strings"
+)
+
+func AppendGitlabSourceToEtree(source *etree.Element, gitSource *devopsv1alpha3.GitlabSource) {
+	source.CreateAttr("class", "io.jenkins.plugins.gitlabbranchsource.GitLabSCMSource")
+	source.CreateAttr("plugin", "gitlab-branch-source")
+	source.CreateElement("id").SetText(gitSource.ScmId)
+	source.CreateElement("serverName").SetText(gitSource.ServerName)
+	source.CreateElement("credentialsId").SetText(gitSource.CredentialId)
+	source.CreateElement("projectOwner").SetText(gitSource.Owner)
+	source.CreateElement("projectPath").SetText(gitSource.Repo)
+	traits := source.CreateElement("traits")
+	if gitSource.DiscoverBranches != 0 {
+		traits.CreateElement("io.jenkins.plugins.gitlabbranchsource.BranchDiscoveryTrait").
+			CreateElement("strategyId").SetText(strconv.Itoa(gitSource.DiscoverBranches))
+	}
+	if gitSource.DiscoverTags {
+		traits.CreateElement("io.jenkins.plugins.gitlabbranchsource.TagDiscoveryTrait")
+	}
+	if gitSource.DiscoverPRFromOrigin != 0 {
+		traits.CreateElement("io.jenkins.plugins.gitlabbranchsource.OriginMergeRequestDiscoveryTrait").
+			CreateElement("strategyId").SetText(strconv.Itoa(gitSource.DiscoverPRFromOrigin))
+	}
+	if gitSource.DiscoverPRFromForks != nil {
+		forkTrait := traits.CreateElement("io.jenkins.plugins.gitlabbranchsource.ForkMergeRequestDiscoveryTrait")
+		forkTrait.CreateElement("strategyId").SetText(strconv.Itoa(gitSource.DiscoverPRFromForks.Strategy))
+		trustClass := "io.jenkins.plugins.gitlabbranchsource.ForkMergeRequestDiscoveryTrait$"
+		switch gitSource.DiscoverPRFromForks.Trust {
+		case 1:
```

Comment:
What does `gitSource.DiscoverPRFromForks.Trust == 1` mean, it's difficult to read.

Reply:
Yes, I agree with you. I didn't give it a better naming base on two reasons. Other git providers doing it the same way. For example, GitHub and Bitbucket. Secondly, This is the original value which comes from Jenkins. I would prefer to refractory them later. How do you think?

Reply:
OK, don't forget

Reply:
In case anyone might interested in it. I address it here https://github.com/kubesphere/kubesphere/pull/3177